--- sqasm.orig.cpp	2009-08-30 16:07:51.000000000 +1000
+++ sqasm.cpp	2009-09-02 22:02:53.000000000 +1000
@@ -328,7 +328,7 @@
 	if( pip>=prog.size() ) return false;
 	
 	string lab;
-	while( getlabel(lab) )
+	while( prog[pip] != '"' && prog[pip] != '\'' && getlabel(lab) )
 	{
 		if( lab2adr.find(lab) == lab2adr.end() )
 			lab2adr[lab] = addr;
@@ -461,7 +461,7 @@
 		string s;
 		getline(cin,s);
 		string::size_type i = s.find("#");
-		if( i!=string::npos ) s = s.substr(0,i);
+		if( i!=string::npos && s[i-1] != '\'' ) s = s.substr(0,i);
 		prog += s+'\n';
 		if( !cin ) break;
 	}
